homeassistant/components/metoffice/weather.py

```diff
@@ -29,22 +41,28 @@ async def async_setup_entry(
     hass_data = hass.data[DOMAIN][entry.entry_id]
 
     async_add_entities(
-        [MetOfficeWeather(entry.data, hass_data,)], False,
+        [
+            MetOfficeWeather(entry.data, hass_data, True),
+            MetOfficeWeather(entry.data, hass_data, False),
+        ],
+        False,
     )
 
 
 class MetOfficeWeather(WeatherEntity):
     """Implementation of a Met Office weather condition."""
 
-    def __init__(self, entry_data, hass_data):
+    def __init__(self, entry_data, hass_data, use_3hourly):
         """Initialise the platform with a data instance."""
         self._data = hass_data[METOFFICE_DATA]
         self._coordinator = hass_data[METOFFICE_COORDINATOR]
 
-        self._name = f"{DEFAULT_NAME} {hass_data[METOFFICE_NAME]}"
-        self._unique_id = f"{self._data.latitude}_{self._data.longitude}"
+        self._name = f"{DEFAULT_NAME} {hass_data[METOFFICE_NAME]} {MODE_3HOURLY_LABEL if use_3hourly else MODE_DAILY_LABEL}"
+        self._unique_id = f"{self._data.latitude}_{self._data.longitude}_{MODE_3HOURLY if use_3hourly else MODE_DAILY}"
```

Comment:
I could, but that struck me as untidy/unbalanced in terms of naming in the short term, but also in the long term it might raise questions as to why only one of the entities had a suffix like that? Which is why I opted to take the hit on a breaking change now, rather than find I needed to later.

Reply:
The `unique_id` isn't seen by the user, only the `name`.  You could append the suffix to both entities for the `name`.  I don't know if it is worth a breaking change to keep the `unique_id` balanced since it is only internal.

Reply:
That's a very good point indeed! Sold! I'll get that done and committed, thanks.

Reply:
It's in, as discussed.